### DISCUSSION TITLE
Lenovo S2 support

### DISCUSSION THREAD

**Discussion by Johnmcenroyy** [InfiniTimeOrg/InfiniTime#1535] [Q&A]
Hello
I have bought Lenovo S2 for experiments with Pinetime's firmwares.

**Lenovo S2**                                
**soc**                               NRF52832                                  
**nor flash**                     XT25F64B (8mb)                            
**display**                        240x240  (ST7789 ?)                          
**touchscreen**               Hynitron CST816S              
**accelerometer**            ST LIS2DW12      (3-axis, temperature)    
**heart rate sensor**       PixArt PAH8011ET (2 green + 1 ir, dsp)       

I have connected with SWD to Openocd-spi with Raspberry Pi.
Unlocked flash with scripts from:
https://github.com/lupyuen/pinetime-updater
Flashed with these scripts then Infinitime bootloader and firmware, but black screen on boot.

According to https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader/pull/10
The P8 watch for example differs from the Pinetime in a few minor points:
   **1. Different pins for the button (Now configurable via libs/pinetime_boot/syscfg.yml)
   2. Some variants have no external low frequency crystal
   3. Different SPI flash chips are used**
   
So I have build fork https://github.com/StarGate01/pinetime-mcuboot-bootloader/tree/p8b
which solves **2** and **3**, flashed it but also black screen.

How I understand I can connect UART (with compiled UART bootloader) to see logs what happens,
but need to search on what pin UART is, for example Pinetime pin TX is number 11.

I have these settings for bootloader:
UART_0:                     1 
UART_0_PIN_TX:     11  # must find correct pin for Lenovo S2 UART
CONSOLE_UART:      1
Is it sufficient for get UART logs or need something else, for example LOG_CLI and LOG_LEVEL settings ?

Any ideas ?
Thanks.

**Comment by Johnmcenroyy** [comment]
It seems that a better way to use Real Time Transfer (RTT) through SWD interface.

**Reply by Avamander** [reply]
Yes, Semihosting or RTT.

**Reply by Johnmcenroyy** [reply]
Hi Avamander.
Does Semihosting will give log of MCUBoot ?
Thanks.

**Reply by Johnmcenroyy** [reply]
Ok, found: https://lupyuen.github.io/pinetime-rust-mynewt/articles/mcuboot#enable-sysinit-hook
Display debugging messages from MCUBoot via the Arm Semihosting Console (in OpenOCD)

**Comment by Johnmcenroyy** [comment]
What is the best way to find unknown pinout of watch to make it support in Infinitime ?
Multimeter, i2c scanner etc. ?

Thanks

**Reply by SteveAmor** [reply]
atc1441 is the expert at reverse engineering these types of watches. A couple of things he has done are a) strip all components off the PCB to check nrf connections and b) intercept companion Android app to get the source of firmware updates, download the firmware update and reverse engineer the code using Ghidra IIRC.

**Reply by Johnmcenroyy** [reply]
Thanks for info. I am read some info at atc1441 github. For b) Unfortunately, I have erased all firmware in flash by deleting APPROTECT security and a) I have just remove all and now check connections to nrf52832. Will see :)

**Comment by ildar** [comment]
I suggest skipping the bootloader for this stage and work with the
InfiniTime fw only (i.e. pinetime-app-image-*.bin).
And to get help on hacking a new device ask at
https://matrix.to/#/!xchlMpnFYulPioyGdI:gitter.im

**Reply by Johnmcenroyy** [reply]
Thank you very much. I just found also that is better for debug without bootloader, 
but for now can't reduce size of firmware for compile with debug enabled.
